### PR TITLE
Added starter theme

### PR DESCRIPTION
**Pull request by bheptinh**
Added STARTER theme

- Moved all theme preprocess functions to `includes` directory
- Added `package-lock.json` file. *Note:* `npm` should be updated to version `^5.0`
- Added `gulpfile.js` and `watchers`
- Added linting for:
   1. scss files - `.scss-lint.yml`
   1. js files - `.jshintrc`

### CODE REVIEW

File: STARTER/scss/style.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+p {
```
Comment (T2L) [inline STARTER/scss/style.scss]: I think we nee to make it just an empty file

File: STARTER/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "scripts": {
+    "test": "echo \"Ecd ../rror: no test specified\" && exit 1",
+    "postinstall": "find node_modules/ -name '*.info' -type f -delete"
+  },
+  "author": "Lemberg Solutions Limited",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/lemberg/draft-theme.git"
+  },
+  "license": "ISC",
```
Comment (T2L) [inline STARTER/package.json]: what's an `ISC`?

File: STARTER/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "starter",
```
Comment (T2L) [inline STARTER/package.json]: Probably should be `STARTER`

File: STARTER/js/src/theme.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (T2L) [inline STARTER/js/src/theme.js:1]: I don't think we need this demo JS code here. Just a template JS file would be enough

File: STARTER/includes/node.inc [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+function STARTER_preprocess_node (&$variables) {
+  // Set our own node type preprocess, drupal creates them by default, but seems to only run hook_preprocess_node__event
+  // on event full pages, not event teasers as well
+  $function = __FUNCTION__ . '_' . $variables['node']->getType();
+  if (function_exists($function)) {
+    $function($variables);
+  }
+}
+
+/**
```
Comment (T2L) [inline STARTER/includes/node.inc]: We do not assume any bundles exist. Please remove

File: STARTER/STARTER.theme [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * Include all files from the includes directory
+ */
+$includes_path = dirname(__FILE__) . '/includes/*.inc';
+foreach (glob($includes_path) as $filename) {
+  require_once dirname(__FILE__) . '/includes/' . basename($filename);
+}
+
+
```
Comment (T2L) [inline STARTER/STARTER.theme]: Extra space

File: STARTER/STARTER.libraries.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+global:
+  version: VERSION
+  css:
+    theme:
+      css/style.css: {}
+  js:
+    js/dist/themekit.js: {}
```
Comment (T2L) [inline STARTER/STARTER.libraries.yml]: I don't like `themekit` here, maybe `draft`. Also, needs to be changed in gulpfile

File: STARTER/STARTER.info.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+name: STARTER
+type: theme
+description: 'A subtheme of Classy, built for STARTER'
+core: 8.x
+# Defines the base theme
+base theme: classy
+
+ckeditor_stylesheets:
+  - css/dist/wysiwyg.css
```
Comment (T2L) [inline STARTER/STARTER.info.yml]: We need to create corresponding scss file, no?

File: STARTER/.scss-lint.yml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  space-before-colon: 1
+  space-after-colon: 1
+  space-before-brace: 1
+  space-before-bang: 1
+  space-after-bang: 1
+  space-between-parens: 1
+
+  # Final Items
+  trailing-semicolon: 1
+  final-newline: 1
```
Comment (T2L) [inline STARTER/.scss-lint.yml]: Missing empty line

File: STARTER/.gitignore [hunk 10]
```diff
@@ -0,0 +1,53 @@
+# Node modules #
```
Comment (T2L) [inline STARTER/.gitignore]: This file contains a lot of stuff. I think it should be project specific, not user machine specific.

**Review by T2L — CHANGES_REQUESTED** [review]
As you have added SCSS and JS lint, please update Travis - it should run those checks!

**Comment by bheptinh** [comment]
@T2L I've fixed all mistakes, you noticed

**Review by andrew-147 — APPROVED** [review]

**Comment by andrew-147** [comment]
@T2L Added newline at the end of files.

**Review by T2L — APPROVED** [review]

**Pull request merged**